hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -411,15 +412,27 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			Type:               string(hyperv1.ValidHostedClusterConfiguration),
 			ObservedGeneration: hcluster.Generation,
 		}
+		isValid := true
+		var msgs []string
 		if err := r.validateConfigAndClusterCapabilities(ctx, hcluster); err != nil {
-			condition.Status = metav1.ConditionFalse
-			condition.Message = err.Error()
-			condition.Reason = hyperv1.InvalidConfigurationReason
-		} else {
+			isValid = false
+			msgs = append(msgs, err.Error())
+		}
+		if err := validateClusterID(hcluster); err != nil {
```

Comment:
any reason why you didn't want to include this check within `validateConfigAndClusterCapabilities` along with validateAzureConfig and validateAzureConfig so no need for the additional isValid tweaks logic?

Reply:
no reason, it makes sense for the check to go in there.
moved it, thx!